app/models/submission.rb

```diff
@@ -32,4 +32,8 @@ class Submission < ApplicationRecord
   def all_entries_valid?
     entries.validated.count == entries.count
   end
+
+  def sheet_names
+    entries.pluck(Arel.sql("source->>'sheet'")).uniq
```

Comment:
The prod DB cut tells me that there could be as many as 319 entries based on current data, so 319 `sheet` values could come back here – can the `Arel.sql` block be made to `GROUP BY source->>'sheet'` so we can drop the in-memory `uniq`?

Reply:
urgh, for some reason I thought this was the way to have it performed in the DB. What I should have used is `distinct`, i.e.

```ruby
entries.distinct.pluck(Arel.sql("source->>'sheet'"))
```

will update 👍🏽

Reply:
Updated